dotnet/src/SemanticKernel/SkillDefinition/NativeFunction.cs

```diff
@@ -738,14 +740,16 @@ private static void TrackUniqueParameterType(ref bool hasParameterType, MethodIn
 
                     // First try to parse using the supplied culture (or current if none was supplied).
                     // If that fails, try with the invariant culture and allow any exception to propagate.
+#pragma warning disable CA1031 // Do not catch general exception types. The code will change in one of the next PRs to allow SK to throw exceptions instead of swallowing them. As a result, this suppression will be removed as well.
                     try
                     {
                         return converter.ConvertFromString(context: null, cultureInfo, input);
                     }
-                    catch (Exception e) when (!e.IsCriticalException() && cultureInfo != CultureInfo.InvariantCulture)
+                    catch (Exception) when (cultureInfo != CultureInfo.InvariantCulture)
```

Comment:
By removing the function `IsCriticalException` we are pratically considering any exceptions to be "critical" (should throw).

Two options: 
1. Remove all the catch block
2. Keep the catch just to log and `throw` after